chia/data_layer/data_store.py

```diff
@@ -350,6 +399,42 @@ async def get_ancestors(self, node_hash: bytes32, tree_id: bytes32, *, lock: boo
 
         return ancestors
 
+    async def get_ancestors_2(
+        self, node_hash: bytes32, tree_id: bytes32, generation: Optional[int] = None, lock: bool = True
+    ) -> List[InternalNode]:
+        nodes = []
+        if generation is None:
+            generation = await self.get_tree_generation(tree_id=tree_id, lock=False)
+        while True:
+            cursor = await self.db.execute(
+                """
+                SELECT * from node INNER JOIN (
```

Comment:
I left this query in non-recursive form for now, as SQL doesn't support aggregates with recursive.